controller/engine_controller.go

```diff
@@ -674,15 +674,27 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 
 	needRestore := false
 	if engine.Status.LastRestoredBackup != "" {
-		info, err := client.Info()
+		rsList, err := client.BackupRestoreStatus()
 		if err != nil {
+			logrus.Errorf("failed to get Backup Restore Status: %v", err)
 			return err
 		}
-		// for those engine just restored from backup, info.LastRestored is empty
-		if info.LastRestored != "" {
-			engine.Status.LastRestoredBackup = info.LastRestored
+
+		for _, status := range rsList {
+			if status.IsRestoring {
+				logrus.Errorf("Engine is currently restoring in replica: %v", status.Replica)
+				return nil
```

Comment:
No... We cannot skip all the remains steps to update engine because one replica is restoring...